pycloudlib/key.py

```diff
@@ -2,13 +2,22 @@
 """Base Key Class."""
 
 import os
+from typing import Optional
+
+from pycloudlib.errors import UnsetSSHKeyError
 
 
 class KeyPair:
     """Key Class."""
 
-    def __init__(self, public_key_path, private_key_path=None, name=None):
-        """Initialize key class to generate key or reuse existing key.
+    def __init__(
+        self,
+        public_key_path: Optional[str],
+        private_key_path: Optional[str] = None,
+        name: Optional[str] = None,
+    ):
+        """
+        Initialize key class to generate key or reuse existing key.
```

Comment:
Nit (applies to more than just here): Docstrings shouldn't start on a newline. https://peps.python.org/pep-0257/#multi-line-docstrings

Reply:
Genuinely how the heck am I just now learning that... I've been doing it this way for 1½ years now lol 

I absolutely LOATHE how that looks but oh well haha